### PR TITLE
Remove obsolete cache subfolder

### PR DESCRIPTION
**Pull request by ansyral**

### CODE REVIEW

File: src/Microsoft.DocAsCode.Build.Engine/DocumentBuilder.cs [hunk 1]
```diff
@@ -189,9 +189,20 @@ public void Build(IList<DocumentBuildParameters> parameters, string outputDirect
...
+                            var root = new DirectoryInfo(Environment.ExpandEnvironmentVariables(_intermediateFolder));
+                            foreach (var dir in root.GetDirectories())
+                            {
+                                if (dir.Name != _currentBuildInfo.DirectoryName)
+                                {
+                                    dir.Delete(true);
+                                }
+                            }
+                        }
+                        catch (Exception ex)
```
Comment (hellosnow) [inline src/Microsoft.DocAsCode.Build.Engine/DocumentBuilder.cs]: >Exception [](start = 31, length = 9)

IOException?

**Comment by hellosnow** [comment]
:shipit:

**Comment by superyyrrzz** [comment]
:shipit:

File: src/Microsoft.DocAsCode.Build.Engine/DocumentBuilder.cs [hunk 2]
```diff
@@ -189,9 +189,20 @@ public void Build(IList<DocumentBuildParameters> parameters, string outputDirect
                     if (_intermediateFolder != null && transformDocument)
                     {
                         _currentBuildInfo.Save(_intermediateFolder);
-                        if (_lastBuildInfo != null)
+                        try
+                        {
+                            var root = new DirectoryInfo(Environment.ExpandEnvironmentVariables(_intermediateFolder));
+                            foreach (var dir in root.GetDirectories())
+                            {
+                                if (dir.Name != _currentBuildInfo.DirectoryName)
```
Comment (vicancy) [inline src/Microsoft.DocAsCode.Build.Engine/DocumentBuilder.cs]: os related case sensitivity?

File: src/Microsoft.DocAsCode.Build.Engine/DocumentBuilder.cs [hunk 3]
```diff
@@ -189,9 +189,20 @@ public void Build(IList<DocumentBuildParameters> parameters, string outputDirect
...
+                                {
+                                    dir.Delete(true);
+                                }
+                            }
+                        }
+                        catch (Exception ex)
                         {
-                            Directory.Delete(Path.Combine(Environment.ExpandEnvironmentVariables(_intermediateFolder), _lastBuildInfo.DirectoryName), true);
+                            Logger.LogVerbose($"Failed to delete obsolete cache files. Details: {ex.Message}.");
```
Comment (vicancy) [inline src/Microsoft.DocAsCode.Build.Engine/DocumentBuilder.cs]: is Verbose too low?

**Comment by ansyral** [comment]
it is a bit risky to delete files that docfx has no idea

**Pull request closed without merging**